### ISSUE TITLE
[BUG] Crash after deleting a job

### ISSUE THREAD

**Issue by dumbasPL** [openbullet/OpenBullet2#166]
**Describe the bug**
Crash after deleting a job

**To Reproduce**
Steps to reproduce the behavior(not exactly sure):
1. Create a job
2. let it rip
3. Go afk
4. delete the job after w while

**Host**
 - OS: Debian 10 x86_64
 - dotnet 5.0.4
 - OpenBullet2 behind reverse proxy (apache2)

**Client**
 - OS: win10
 - Browser: chrome 89

**Additional context**

### log 
```
-- Logs begin at Mon 2021-03-22 21:58:11 CET, end at Mon 2021-03-22 22:56:30 CET. --
Mar 22 22:54:24 nezu dotnet[31042]: warn: Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer[100]
Mar 22 22:54:24 nezu dotnet[31042]:       Unhandled exception rendering component: A task was canceled.
Mar 22 22:54:24 nezu dotnet[31042]:       System.Threading.Tasks.TaskCanceledException: A task was canceled.
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.RenderTree.Renderer.<RenderInExistingBatch>g__GetHandledAsynchronousDisposalErrorsTask|48_0(Task result)
Mar 22 22:54:24 nezu dotnet[31042]: fail: Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
Mar 22 22:54:24 nezu dotnet[31042]:       Unhandled exception in circuit 'Spt6Ln4GiYRhquvCE0Kf0Jdszepd0qkIpQI3MXT87QI'.
Mar 22 22:54:24 nezu dotnet[31042]:       System.Threading.Tasks.TaskCanceledException: A task was canceled.
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntimeExtensions.InvokeVoidAsync(IJSRuntime jsRuntime, String identifier, Object[] args)
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Virtualization.VirtualizeJsInterop.DisposeAsync()
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Virtualization.Virtualize`1.DisposeAsync()
Mar 22 22:54:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.RenderTree.Renderer.<RenderInExistingBatch>g__GetHandledAsynchronousDisposalErrorsTask|48_0(Task result)
Mar 22 22:55:24 nezu dotnet[31042]: warn: Microsoft.AspNetCore.Components.Server.Circuits.RemoteRenderer[100]
Mar 22 22:55:24 nezu dotnet[31042]:       Unhandled exception rendering component: A task was canceled.
Mar 22 22:55:24 nezu dotnet[31042]:       System.Threading.Tasks.TaskCanceledException: A task was canceled.
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Extensions.Head.HeadManagementJSObjectReference.RemoveHeadTagAsync(String id)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Extensions.Head.HeadTagBase.DisposeAsync()
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
Mar 22 22:55:24 nezu dotnet[31042]: fail: Microsoft.AspNetCore.Components.Server.Circuits.CircuitHost[111]
Mar 22 22:55:24 nezu dotnet[31042]:       Unhandled exception in circuit 'Spt6Ln4GiYRhquvCE0Kf0Jdszepd0qkIpQI3MXT87QI'.
Mar 22 22:55:24 nezu dotnet[31042]:       System.Threading.Tasks.TaskCanceledException: A task was canceled.
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSRuntime.InvokeAsync[TValue](Int64 targetInstanceId, String identifier, Object[] args)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.JSInterop.JSObjectReferenceExtensions.InvokeVoidAsync(IJSObjectReference jsObjectReference, String identifier, Object[] args)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Extensions.Head.HeadManagementJSObjectReference.RemoveHeadTagAsync(String id)
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.Web.Extensions.Head.HeadTagBase.DisposeAsync()
Mar 22 22:55:24 nezu dotnet[31042]:          at Microsoft.AspNetCore.Components.RenderTree.Renderer.<Dispose>g__HandleAsyncExceptions|52_0(List`1 tasks)
Mar 22 22:55:25 nezu dotnet[31042]: Unhandled exception. System.NullReferenceException: Object reference not set to an instance of an object.
Mar 22 22:55:25 nezu dotnet[31042]:    at OpenBullet2.Shared.MultiRunJobViewer.SaveJobOptions(Object sender, EventArgs e) in D:\a\OpenBullet2\OpenBullet2\OpenBullet2\Shared\MultiRunJobViewer.razor.cs:line 144
Mar 22 22:55:25 nezu dotnet[31042]:    at System.Threading.ExecutionContext.RunInternal(ExecutionContext executionContext, ContextCallback callback, Object state)
Mar 22 22:55:25 nezu dotnet[31042]: --- End of stack trace from previous location ---
Mar 22 22:55:25 nezu dotnet[31042]:    at System.Threading.TimerQueueTimer.CallCallback(Boolean isThreadPool)
Mar 22 22:55:25 nezu dotnet[31042]:    at System.Threading.TimerQueueTimer.Fire(Boolean isThreadPool)
Mar 22 22:55:25 nezu dotnet[31042]:    at System.Threading.TimerQueue.FireNextTimers()
Mar 22 22:55:25 nezu systemd[1]: openbullet2.service: Main process exited, code=killed, status=6/ABRT
Mar 22 22:55:25 nezu systemd[1]: openbullet2.service: Failed with result 'signal'.
```

**Comment by openbullet** [comment]
Hi, which version of OB2 are you using? Is it the latest (0.1.8)?

**Comment by meinname** [comment]
Tried it with multiple jobs on 0.1.8.
Was not able to reproduce.

**Comment by dumbasPL** [comment]
> Hi, which version of OB2 are you using? Is it the latest (0.1.8)?

yes 0.1.8  
I was hoping the stack trace might help since I expected an almost "random" crash to be hard to reproduce.
Is there any debugger I can attach to it(preferably a remote one) that will capture the crash if it ever happens again? I never debugged dotnet on a Linux system.

**Comment by openbullet** [comment]
The stack trace you provided was actually pretty useful, I got mislead by the `TaskCanceledException` but that is just some js-related noise and can safely be ignored. The actual exception was the last one (`NullReferenceException` at `MultiRunJobViewer.razor.cs:line 144`). Hopefully the commit I pushed will solve this issue. Also a rework of that section is planned.
Reopen if the issue is still present.
